test/DynamoCoreWpfTests/ConnectorViewModelTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿using System.Linq;
+using System.Windows.Input;
+using Dynamo.Selection;
+using DynamoCoreWpfTests.Utility;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.IO;
+using Dynamo.Graph.Workspaces;
+using static Dynamo.Models.DynamoModel;
+using Dynamo.Scheduler;
+using Dynamo.Utilities;
+
+namespace DynamoCoreWpfTests
+{
+    public class ConnectorViewModelTests: DynamoTestUIBase
+    {
+
+        #region Regular Connector Tests
+        /// <summary>
+        /// Check to see a pin can be added to a connector
+        /// </summary>
+        [Test]
+        public void CanPinConnector()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+
+            ///hard-coded position (1) on wire
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 1);
+
+            ///hard-coded position (2) on wire
+            connectorViewModel.PanelX = 244;
+            connectorViewModel.PanelY = 303.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 2);
+        }
+
+        /// <summary>
+        /// Check to see a pin can be removed from a connector
+        /// </summary>
+        [Test]
+        public void CanUnPinFromConnector()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+
+            ///hard-coded position (1) on wire
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+
+            ///Request remove pin
+            var firstPin = connectorViewModel.ConnectorPinViewCollection.First();
+            firstPin.OnRequestRemove(firstPin, System.EventArgs.Empty);
+
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 0);
```

Comment:
in cases like this, it is generally good to assert that something has actually been removed, this test doesn't actually tell us if it was removed as we dont know what ConnectorPinViewCollection.Count was before we did remove

Reply:
fyi this goes for most of the tests in this class

Reply:
Cool, modified all tests in this class to assert more meaningful things :).